src/cyclebane/graph.py

```diff
@@ -430,7 +430,8 @@ def __setitem__(self, branch: Hashable | slice, other: Graph) -> None:
         intersection_nodes = set(graph.nodes) & set(new_branch.nodes) - {branch}
 
         for node in intersection_nodes:
-            if graph.pred[node] != new_branch.pred[node]:
+            new_pred = new_branch.pred[node]
+            if new_pred and graph.pred[node] != new_pred:
```

Comment:
More generally, would it make sense to check if all keys in "new" have the same values in existing, basically if a dict update would have no effect?